### PR TITLE
Silence Xcode 8.3 warnings

### PR DESCRIPTION
**Pull request by steam**
For the first time, in a very long time, we have zero warnings!

screen shot 2017-03-28 at 9 06 28 am

### CODE REVIEW

**Comment by jayzes** [comment]
🙈

**Comment by steam** [comment]
>Executed 2568 tests, with 0 failures (0 unexpected) in 235.203 (236.236) seconds

**Comment by steam** [comment]
The no Buildtime Issues image is a **LIE**. But, we're back to the known ones related to yap nullability specifiers and the one `setStatusBarHidden` warning in the specs.

File: Sources/Controllers/Profile/ProfileViewController.swift [hunk 1]
```diff
@@ -207,7 +207,7 @@ final class ProfileViewController: StreamableViewController {
         let gridListItem = UIBarButtonItem.gridListItem(delegate: streamViewController, isGridView: streamViewController.streamKind.isGridView)
         let shareItem = UIBarButtonItem(image: .share, target: self, action: #selector(ProfileViewController.sharePostTapped(_:)))
         let moreActionsItem = UIBarButtonItem(image: .dots, target: self, action: #selector(ProfileViewController.moreButtonTapped))
-        let isCurrentUser = userParam == currentUser?.id || userParam == "~\(currentUser)"
+        let isCurrentUser = userParam == currentUser?.id || userParam == "~\(String(describing: currentUser))"
```
Comment (colinta) [inline Sources/Controllers/Profile/ProfileViewController.swift]: This `String(describing: )` code is a nuisance 🙁

File: Sources/Controllers/Stream/Cells/StreamHeaderCell.swift [hunk 2]
```diff
@@ -489,7 +489,7 @@ extension StreamHeaderCell {
             rotateChevron(CGFloat(0))
         }
         else {
-            rotateChevron(CGFloat(M_PI))
+            rotateChevron(CGFloat(Double.pi))
```
Comment (colinta) [inline Sources/Controllers/Stream/Cells/StreamHeaderCell.swift]: Is there a `CGFloat.pi`?

File: Sources/Controllers/Profile/ProfileViewController.swift (hunk 1 above)
Comment (steam) [inline Sources/Controllers/Profile/ProfileViewController.swift]: Yes, total noise. :(

File: Sources/Controllers/Stream/Cells/StreamHeaderCell.swift (hunk 2 above)
Comment (steam) [inline Sources/Controllers/Stream/Cells/StreamHeaderCell.swift]: Hey! Just checked, yep. I'll use that were I can. :)

File: Sources/Utilities/Tmp.swift [hunk 3]
```diff
@@ -63,8 +63,11 @@ struct Tmp {
     }
 
     static func read(_ fileName: String) -> String? {
-        if let data: Data = read(fileName) {
-            return NSString(data: data, encoding: String.Encoding.utf8.rawValue) as? String
+        if
+            let data: Data = read(fileName),
+            let string = NSString(data: data, encoding: String.Encoding.utf8.rawValue)
```
Comment (colinta) [inline Sources/Utilities/Tmp.swift]: There's a swift version of this: `String(data: data, encoding: .utf8)`

**Pull request merged**